### PR TITLE
chore: fix warnings on nightly

### PR DESCRIPTION
**Pull request by davidbarsky**
This PR fixes two types of warnings:
1. Usage of deprecated methods on `std::error::Error`.
2. An unused warning on `extern crate proc_macro` in tracing-attributes. This does not occur on the stable release of `1.41`, but it does occur on `1.42.0-beta.1` and `1.42.0-nightly (cd1ef390e 2020-01-31)`. 

Prior to merging this PR, I'd like to determine whether the second warning is a deliberate change or a regression.

### CODE REVIEW

**Comment by davidbarsky** [comment]
It appears to be deliberate; I'm now able to build `tracing-attributes` without `extern crate proc_macro` on beta and nightly.

**Comment by davidbarsky** [comment]
@mystor informed me that this _is_ a deliberate change in Cargo (https://github.com/rust-lang/cargo/pull/7700). Thanks!

File: tracing-subscriber/src/filter/env/mod.rs [hunk 1]
```diff
@@ -406,20 +406,13 @@ impl From<env::VarError> for FromEnvError {
 impl fmt::Display for FromEnvError {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self.kind {
-            ErrorKind::Parse(ref p) => p.fmt(f),
-            ErrorKind::Env(ref e) => e.fmt(f),
```
Comment (hawkw) [inline tracing-subscriber/src/filter/env/mod.rs]: what's the motivation for this change? it doesn't seem like `<impl Display>::fmt` should warn?

File: tracing-subscriber/src/reload.rs [hunk 2]
```diff
@@ -228,15 +228,12 @@ impl Error {
...
-
-impl error::Error for Error {
-    fn description(&self) -> &str {
-        match self.kind {
+        let msg = match self.kind {
             ErrorKind::SubscriberGone => "subscriber no longer exists",
             ErrorKind::Poisoned => "lock poisoned",
-        }
+        };
+        write!(f, "{}", msg)
```
Comment (hawkw) [inline tracing-subscriber/src/reload.rs]: slight pref for
```suggestion
        f.pad(msg)
```
there is _slight_ overhead from evaluating format macros that can be avoided when we aren't actually interpolating

File: tracing-subscriber/src/filter/env/mod.rs (hunk 1 above)
Comment (davidbarsky) [reply tracing-subscriber/src/filter/env/mod.rs]: Blind copy & paste!
